easybuild/tools/module_generator.py

```diff
@@ -965,6 +981,10 @@ def update_paths(self, key, paths, prepend=True, allow_abs=False, expand_relpath
         :param allow_abs: allow providing of absolute paths
         :param expand_relpaths: expand relative paths into absolute paths (by prefixing install dir)
         """
+        paths = self._filter_paths(key, paths)
+        if paths is None:
```

Comment:
I was wondering about how the filtered paths could be `None`:   
This is inside `_filter_paths`:
```
if not filtered_paths:
    filtered_paths = None
```

I would remove both checks, neither is required and we are actually checking stuff twice. Having `_filter_paths` always return a list, even if it is empty is cleaner interface-wise. I mean otherwise you would need to document it as "returns a non-empty list of paths to add or None" instead of "returns a list of paths to add".

If we really want to exit early here, then it is better to remove the above check in `_filter_paths` and change this to:
```suggestion
        if not paths:
```

That would at least check only once